test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -90,24 +92,33 @@ AndroidArgs
     }
 }
 
+// Changes these based on type
 val AndroidArgs.isDontAutograntPermissions
     get() = !(grantPermissions.isNotNull() && (grantPermissions.equals("all")))
 
 val AndroidArgs.isInstrumentationTest
-    get() = appApk.isNotNull() &&
-            testApk.isNotNull() ||
-            additionalAppTestApks.isNotEmpty() &&
-            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() })
+    get() = if (type != null) (type == Type.INSTRUMENTATION) else validateInstrumentation()
+
+fun AndroidArgs.validateInstrumentation() = (appApk.isNotNull() && testApk.isNotNull() ||
+        additionalAppTestApks.isNotEmpty() &&
+        (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() }))
 
 val AndroidArgs.isRoboTest
+    get() = if (type != null) type == Type.ROBO else validateRobo()
+
+fun AndroidArgs.validateRobo() = (appApk.isNotNull() && (roboDirectives.isNotEmpty() || roboScript.isNotNull()))
+
+val AndroidArgs.checkForSanityRobo
```

Comment:
I think `checkForSanityRobo` is not used anywhere else, let's make it private

Reply:
👍